src/document/DocumentCommandHandlers.js

```diff
@@ -1596,6 +1597,12 @@ define(function (require, exports, module) {
         ProjectManager.showInTree(MainViewManager.getCurrentlyViewedFile(MainViewManager.ACTIVE_PANE));
     }
 
+    /** Download selected file or folder structure **/
+    function handleFileDownload() {
+        var entry = ProjectManager.getSelectedItem();
+        ArchiveUtils.archiveWithPath(entry._path,function(){});
```

Comment:
Depending on what this is (e.g., file vs. folder) I think we should do different things.  I think it's odd to click "Download" on a file and have it send me a `.zip`.  What I'd do is check to see if this is a file vs. folder, and based on that do one of two things:

* For the file case: generalize my code here https://github.com/simon66/brackets/blob/70a2290e4e64ce5e49fe3a3398c11c852d2939f2/src/filesystem/impls/filer/ArchiveUtils.js#L208-L210 to have it get the blob URL from `BlobUtils` for this path and then use that with `saveAs()`.  Then you'll just get the actual file you want.
* If it's a dir, I think archiving is probably the right move, but we should probably alter what we send.  Fixing the "project/" bug here might make sense, since we should just send the folder zipped.

I also don't love how you're passing an empty callback.  It would be better to check for `callback` in `archiveWithPath` and if it's not there, create it then.